### PR TITLE
8320943: Files/probeContentType/Basic.java fails on latest Windows 11 - content type mismatch

### PR DESCRIPTION
**Pull request by amosshi**
Backport of [JDK-8320943](https://bugs.openjdk.org/browse/JDK-8320943)
- This PR has two commits
  - commit 1 is the backport of the original [commit](https://github.com/openjdk/jdk/commit/87516e29dc5015c4cab2c07c5539ad30f2768667)
  - commit 2 is fixing the compile and runtime issue on Java 17.
    - Compile issue fixed: on Java 17 the class is `jdk.test.lib.Platform` and `jdk.test.lib.OSVersion`
    - Runtime issue fixed: on Java 17 there is no mime `wasm`, so we comment it out
      - `wasm` exists in Java 21: https://github.com/openjdk/jdk21u-dev/blob/master/src/java.base/windows/classes/sun/net/www/content-types.properties
      - `wasm` does not exists in Java 17: https://github.com/openjdk/jdk17u-dev/blob/master/src/java.base/windows/classes/sun/net/www/content-types.properties 
    - :white_check_mark: after [JDK-8297609](https://bugs.openjdk.org/browse/JDK-8297609) back ported to 17, `wasm` will be available
- So this is an unclean backport

Testing
- Local: Test passed on Mac
  - `Basic.java` - Test results: passed: 1

Mac machine

```
MacBook Pro
16-inch, 2021
Chip: Apple M1 Max
MacOS: 14.3.1 (23D60)
```

Windows machine: 

```
Device name	W-PW05T4AD
Processor	12th Gen Intel(R) Core(TM) i7-12800H   2.40 GHz
Installed RAM	64.0 GB (63.7 GB usable)
System type	64-bit operating system, x64-based processor
```

- :yellow_heart: Pipeline: All `GHA Sanity Checks` passed except `linux-cross-compile / build (riscv64)`
  - :x: It failed with message `Error: Unable to find an artifact with the name: bundles-linux-x64`
  - :white_check_mark: This error is on Linux riscv64 - which not caused by current test case, because current PR is for `Windows`

- Testing Machine: SAP nightlies passed on `2024-03-01,02,03`

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8320943](https://bugs.openjdk.org/browse/JDK-8320943) needs maintainer approval

### Issue
 * [JDK-8320943](https://bugs.openjdk.org/browse/JDK-8320943): Files/probeContentType/Basic.java fails on latest Windows 11 - content type mismatch (**Bug** - P4 - Approved)

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev.git pull/2254/head:pull/2254` \
`$ git checkout pull/2254`

Update a local copy of the PR: \
`$ git checkout pull/2254` \
`$ git pull https://git.openjdk.org/jdk17u-dev.git pull/2254/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2254`

View PR using the GUI difftool: \
`$ git pr show -t 2254`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/2254.diff">https://git.openjdk.org/jdk17u-dev/pull/2254.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk17u-dev/pull/2254#issuecomment-1970104903)

### CODE REVIEW

File: test/jdk/java/nio/file/Files/probeContentType/Basic.java [hunk 1]
```diff
@@ -153,39 +156,55 @@ public static void main(String[] args) throws IOException {
...
+        exTypes.add(new ExType("php", List.of("text/plain", "text/php", "application/x-php")));
+        exTypes.add(new ExType("png", List.of("image/png")));
+        exTypes.add(new ExType("ppt", List.of("application/vnd.ms-powerpoint")));
+        exTypes.add(new ExType("pptx", List.of("application/vnd.openxmlformats-officedocument.presentationml.presentation")));
+        exTypes.add(new ExType("py", List.of("text/plain", "text/x-python", "text/x-python-script")));
+        exTypes.add(new ExType("webm", List.of("video/webm")));
+        exTypes.add(new ExType("webp", List.of("image/webp")));
+        exTypes.add(new ExType("xls", List.of("application/vnd.ms-excel")));
+        exTypes.add(new ExType("xlsx", List.of("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")));
+        // exTypes.add(new ExType("wasm", List.of("application/wasm")));
```
Comment (TheRealMDoerr) [inline test/jdk/java/nio/file/Files/probeContentType/Basic.java:188]: Ok, [JDK-8297609](https://bugs.openjdk.org/browse/JDK-8297609) is not in 17u. Can you add a comment explaining this, please?

**Review by TheRealMDoerr — COMMENTED** [review]
LGTM.

File: test/jdk/java/nio/file/Files/probeContentType/Basic.java (hunk 1 above)
Comment (amosshi) [reply test/jdk/java/nio/file/Files/probeContentType/Basic.java]: @TheRealMDoerr - Done

**Review by TheRealMDoerr — APPROVED** [review]
I would have spent a separate line, but ok.

**Comment by amosshi** [comment]
/approval request "Backporting for parity with 17.0.12-oracle. No risk, test fix. Un-Clean backport. SAP nightlies passed on 2024-03-01,03”

**Pull request closed without merging**